### ISSUE TITLE
Unable to use auth Action with GKE Workload Identity

### ISSUE THREAD

**Issue by kevholmes** [google-github-actions/auth#293]
### TL;DR

I am running self-hosted GitHub Actions runners in GKE. The GKE runner cluster has Workload Identity enabled. I am mapping Actions runners to kubernetes service accounts that map to GCP IAM service accounts in the same project and hoping to get an access_token token to use in later Jobs. However, I get an error for invalid audience when trying this.

I was hoping to be able to do this:
```yaml
    - id: 'auth'
      name: 'Authenticate to Google Cloud'
      uses: 'google-github-actions/auth@v1'
      with:
        workload_identity_provider: 'cluster-name.svc.id.goog'
        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
```

### Expected behavior

I was hoping this would allow me to obtain a token I could then use to auth against things like GCR:

```yaml
      - name: Authenticate with gcr.io
        uses: docker/login-action@v2
        with:
          registry: gcr.io
          username: oauth2accesstoken
          password: ${{ needs.gcp-auth.outputs.auth-token }}
```

### Observed behavior

Currently, I can query the metadata service within my Actions runner GKE cluster to obtain an token with a timeout of one hour per the service accounts bound to my runner pods and use it to auth against other google services manually.

### Action YAML

```yaml
jobs:
  gcp-auth:
    runs-on: runner-1
    outputs:
      auth-token: ${{ steps.auth.outputs.access_token }}
    permissions:
      contents: 'read'
      id-token: 'write'
    steps:
      - name: 'Authenticate to Google Cloud'
        id: 'auth'
        uses: external-actions/auth@v1
        with:
          token_format: access_token
          workload_identity_provider: 'cluster-name.svc.id.goog'
          service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
  buildx:
    needs: gcp-auth
    runs-on: runner-1
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Set up Docker Buildx
        uses: external-actions/setup-buildx-action@v2
      - name: Authenticate with gcr.io
        uses: external-actions/login-action@v2
        with:
          registry: gcr.io
          username: oauth2accesstoken
          password: ${{ needs.gcp-auth.outputs.auth-token }}
```

### Log output

```text
"error_description":"Invalid value for \"audience\". This value should be the full resource name of the Identity Provider.
```

**Comment by sethvargo** [comment]
Hi @kevholmes [Workload Identity Federation](https://cloud.google.com/iam/docs/workload-identity-federation) is separate from [GKE Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity). Querying the metadata server directly (or using `gcloud`) is the correct approach for what you're doing.
